dogfood_tests/test_checkov_dogfood.py

```diff
@@ -0,0 +1,171 @@
+    ]
+
+    run_framework_test(caplog=caplog, framework=CheckType.CLOUDFORMATION, excluded_paths=excluded_paths)
+
+
+def test_dockerfile_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.DOCKERFILE)
+
+
+def test_github_actions_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.GITHUB_ACTIONS)
+
+
+def test_gitlab_ci_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.GITLAB_CI)
+
+
+def test_helm_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.HELM)
+
+
+def test_kubernetes_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.KUBERNETES)
+
+
+@pytest.mark.skip(reason="kustomize needs a context to do a proper scan, which is hard to set here")
+def test_kustomize_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.KUSTOMIZE)
+
+
+def test_openapi_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.OPENAPI)
+
+
+def test_secrets_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.SECRETS)
+
+
+def test_serverless_framework(caplog: LogCaptureFixture) -> None:
+    run_framework_test(caplog=caplog, framework=CheckType.SERVERLESS)
+
+
+def test_terraform_framework(caplog: LogCaptureFixture) -> None:
+    excluded_paths = [
+        "terraform/runner/resources/example/invalid.tf$",
+        "terraform/runner/resources/invalid_terraform_syntax/bad_tf_1.tf$",
+        "terraform/runner/resources/invalid_terraform_syntax/bad_tf_2.tf$",
+        "terraform/runner/resources/unbalanced_eval_brackets/main.tf$",
+    ]
+
+    run_framework_test(caplog=caplog, framework=CheckType.TERRAFORM, excluded_paths=excluded_paths)
+
+
+def test_terraform_plan_framework(caplog: LogCaptureFixture) -> None:
+    excluded_paths = [
+        "cloudformation/parser/fail.json$",
+        "cloudformation/parser/success_triple_quotes_string.json$",
+        "cloudformation/runner/resources/invalid.json$",
+    ]
+
+    run_framework_test(caplog=caplog, framework=CheckType.TERRAFORM_PLAN, excluded_paths=excluded_paths)
+
+
+def run_framework_test(caplog: LogCaptureFixture, framework: str, excluded_paths: list[str] | None = None) -> None:
+    # given
+    caplog.set_level(logging.ERROR)
+    runner_registry = RunnerRegistry(
+        "", RunnerFilter(framework=[framework], excluded_paths=excluded_paths), *DEFAULT_RUNNERS
+    )
+
+    # when
+    scan_reports = runner_registry.run(root_folder=str(TESTS_DIR))
+
+    # then
+    for report in scan_reports:
+        assert report.failed_checks
+        assert not report.parsing_errors, f"Found parsing errors for framework '{report.check_type}'"
+
+    assert not caplog.text, caplog.text
```

Comment:
the second `caplog.text` is a typo?

Reply:
actually not 😄 if it fails it will print out the complete caplog text, just wanted to make sure, nothing is cut off